jwst/flatfield/flat_field.py

```diff
@@ -301,34 +301,47 @@ def do_NIRSpec_flat_field(output_model,
         xstop = xstart + xsize
         ystop = ystart + ysize
 
-        # Make sure there is a WCS.
-        if not hasattr(slit.meta, "wcs") or slit.meta.wcs is None:
-            log.error("Slit %s does not have a 'wcs' attribute.", slit.name)
-            if output_model.meta.cal_step.assign_wcs == 'COMPLETE':
-                raise RuntimeError("WCS was not found, but it isn't clear "
-                                   "why not.")
+        got_wcs = hasattr(slit.meta, "wcs") and slit.meta.wcs is not None
+
+        # Get the wavelength at each pixel in the extracted slit data.
+        wl = slit.wavelength                    # a 2-D array
```

Comment:
Would this raise an AtrttibuteError in slitless mode, since there's no wavelength array written out? Should the code check if the `wavelength` attribute is available? Or is the `wavelength` attribute always available because of the schema?

Reply:
This section would only be executed for `MultiSlit` data, so the attribute should always exist because of the schema, but I'll add a check anyway.  It's better not to rely too much on assumptions.